src/Common/FileSyncGuard.cpp

```diff
@@ -0,0 +1,36 @@
+#include <Common/FileSyncGuard.h>
+#include <Common/Exception.h>
+#include <Disks/IDisk.h>
+#include <fcntl.h> // O_RDWR
+
+/// OSX does not have O_DIRECTORY
+#ifndef O_DIRECTORY
+#define O_DIRECTORY O_RDWR
+#endif
+
+namespace DB
+{
+
+FileSyncGuard::FileSyncGuard(const DiskPtr & disk_, const String & path)
```

Comment:
The name of `FileSyncGuard` class does not reflect the fact that it works only on directories.